test/test_multidevice_pipeline.cpp

```diff
@@ -304,6 +304,122 @@ INSTANTIATE_TEST_SUITE_P(
         ::testing::Values(true),
         ::testing::Bool()));
 
+enum class SchedulingMode {
+  noIntraDeviceScheduling,
+  manualScheduling,
+};
+
+std::ostream& operator<<(std::ostream& out, const SchedulingMode& mode) {
```

Comment:
Sounds like a useful utility that should belong to csrc/type.h?

Reply:
Don't you expect it to be shortly used elsewhere? For those types, I tend to move them to a common header, but I also don't mind keeping them local for now and refactor later.

Reply:
I don't know exactly how it could be used later, so I'd rather keep it here and refactor when needed